pipeline_components/_tensorflow/_tensorflow/train/component.yaml

```diff
@@ -0,0 +1,458 @@
+    Attribute .path is the GCS location for csv files
+- name: validation_data
+  type: Dataset
+  description: |-
+    Validation data as kfp's Dataset object.
+    Attribute .path is the GCS location for csv files
+- name: file_pattern
+  type: String
+  description: |-
+    Read data from one or more files. If empty, then
+    training and validation data is read from single file respectively.
+    For multiple files, use a pattern e.g. "files-*.csv".
+- {name: label_name, type: String, description: Name of column containing the labels.}
+- name: model_params
+  type: JsonObject
+  description: |-
+    Dictionary of following training parameters
+    batch_size: int (defaults to 100)
+    epochs: int (defaults to 5)
+    learning rate: float (defaults to 0.001)
+    hidden_units: list(tuple) (defaults to [(10, 'relu')])
+        Example - [(64, "relu"), (32, "elu")]
+        creates 1st dense layer with 64 nodes & activation function as relu
+        & 2nd dense layer with 32 nodes & activation function as elu
+        Reference (activation functions):
+            https://www.tensorflow.org/api_docs/python/tf/keras/activations
+    loss_fn: str (defaults to MeanSquaredError)
+        Regression:
+            MeanSquaredError, MeanAbsoluteError, MeanAbsolutePercentageError
+        Classification:
+            BinaryCrossentropy,
+            CategoricalCrossentropy, SparseCategoricalCrossentropy
+        Reference:
+            https://www.tensorflow.org/api_docs/python/tf/keras/losses
+    optimizer: str (defaults to Adam)
+        Supported values:
+            Adam, Adadelta, Adamax, Adagrad,
+            Ftrl, RMSprop, SGD
+        Reference:
+            https://www.tensorflow.org/api_docs/python/tf/keras/optimizers
+    metrics: list (defaults to ['Accuracy'])
+        Reference:
+            https://www.tensorflow.org/api_docs/python/tf/keras/metrics
+    distribute strategy: str (defaults to single)
+        Supported values:
+            Without GPU: single
+            With GPU: single, mirror, multi
+    early_stopping_epochs: int (defaults to 5)
+        No of epochs to check for training loss convergence
+outputs:
+- name: model
+  type: Model
+  description: |-
+    Output model as a kfp Model object.
+    Attribute .path is the GCS location for the trained model
+    binaries in protobuf format
+- name: metrics_artifact
+  type: Artifact
+  description: |-
+    Output metrics of all iterations for
+    the trained model in JSON format.
+implementation:
+  container:
+    image: python:3.7
+    command:
+    - sh
+    - -c
+    - |2
+
+      if ! [ -x "$(command -v pip)" ]; then
+          python3 -m ensurepip || python3 -m ensurepip --user || apt-get install python3-pip
+      fi
+
+      PIP_DISABLE_PIP_VERSION_CHECK=1 python3 -m pip install --quiet     --no-warn-script-location 'tensorflow==2.7.1' 'kfp==1.8.9' && "$0" "$@"
+    - sh
+    - -ec
+    - |
+      program_path=$(mktemp -d)
```

Comment:
This much code shouldn't be in yaml. It makes it very hard to CR, test or develop on. Same for all `component.yaml` files

Reply:
removed component.yaml files